sled-agent/src/services.rs

```diff
@@ -479,6 +456,60 @@ impl ServiceManager {
         Ok(None)
     }
 
+    // Check the services intended to run in the zone to determine whether any
+    // OPTE ports need to be created and mapped into the zone when it is created.
+    fn opte_ports_needed(
+        &self,
+        req: &ServiceZoneRequest,
+    ) -> Result<Vec<(Port, PortTicket)>, Error> {
+        let port_manager = &self
+            .inner
+            .sled_info
+            .get()
+            .ok_or_else(|| Error::SledAgentNotReady)?
+            .port_manager;
+
+        let mut ports = vec![];
+        for svc in &req.services {
+            match svc {
+                ServiceType::Nexus { external_ip, .. } => {
+                    // TODO: Arbitrary MAC choice from the "system" portion of the
+                    // virtual MAC address space.
+                    let mac = [0xA8, 0x40, 0x25, 0xFF, 0x00, 0x01].into();
+                    // TODO: Arbitrary VNI choice from Oxide-reserved range.
+                    let vni = 100;
+                    // Allow incoming web traffic for nexus
+                    // TODO: Should come froma service VPC (along with the VNI).
+                    let firewall_rules = &[serde_json::from_str(
```

Comment:
why not use the type here?

Reply:
Just was a bit more annoying to fill in when I first wrote it :P Anyways, this bits been removed and moved elsewhere.